rust/cardano-chain-follower/src/stats/mmap_file.rs

```diff
@@ -0,0 +1,88 @@
+//! Memory-mapped file statistics.
+
+use core::fmt;
+use std::sync::{
+    atomic::{AtomicBool, AtomicU64, Ordering},
+    Arc,
+};
+
+use serde::{ser::SerializeStruct, Serialize, Serializer};
+
+/// Memory-mapped file statistics.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct MemMapFileStat(Arc<InnerMemMapFileStat>);
+
+/// Inner memory-mapped file statistics.
+struct InnerMemMapFileStat {
+    /// A counter for the number of memory-mapped files.
+    file_counter: AtomicU64,
+    /// The total size of memory-mapped files.
+    total_size: AtomicU64,
+    /// A boolean value indicating whether the memory-mapped files have been dropped.
+    is_drop: AtomicBool,
```

Comment:
A bool is insufficient.
Have:
```
drop_count: AtominU64,
drop_size: AtomicU64,
memmap_errors: AtomicU64,
```
For generality, this type should probably be in `cat-types`,  and it should internally maintain its own stats.
So cat-gateway can just use `MemoryMapFile::stats()` to get the current stats.
Then the stat would be global, in case we use memmap files in other libraries or cat-gateway itself.

Just keep the stats as static lazy_alloc inside the type itself.
This code here then could have `InnerMemMapFileStat` be empty, and the Serialize just uses the valuse directly from MemoryMapFile type itself.